ENIGMAsystem/SHELL/Widget_Systems/General/WSdialogs.h

```diff
@@ -22,71 +23,74 @@ using std::string;
 };
 
 // These are constants defined inside one of the win32 headers in mingw, no need to define them 2x
 /*
 enum {
-	ANSI_CHARSET,
-	DEFAULT_CHARSET,
-	EASTEUROPE_CHARSET,
-	RUSSIAN_CHARSET,
-	SYMBOL_CHARSET,
-	SHIFTJIS_CHARSET,
-	HANGEUL_CHARSET,
-	GB2312_CHARSET,
-	CHINESEBIG5_CHARSET,
-	JOHAB_CHARSET,
-	HEBREW_CHARSET,
-	ARABIC_CHARSET,
-	GREEK_CHARSET,
-	TURKISH_CHARSET,
-	VIETNAMESE_CHARSET,
-	THAI_CHARSET,
-	MAC_CHARSET,
-	BALTIC_CHARSET,
-	OEM_CHARSET
+  ANSI_CHARSET,
+  DEFAULT_CHARSET,
+  EASTEUROPE_CHARSET,
+  RUSSIAN_CHARSET,
+  SYMBOL_CHARSET,
+  SHIFTJIS_CHARSET,
+  HANGEUL_CHARSET,
+  GB2312_CHARSET,
+  CHINESEBIG5_CHARSET,
+  JOHAB_CHARSET,
+  HEBREW_CHARSET,
+  ARABIC_CHARSET,
+  GREEK_CHARSET,
+  TURKISH_CHARSET,
+  VIETNAMESE_CHARSET,
+  THAI_CHARSET,
+  MAC_CHARSET,
+  BALTIC_CHARSET,
+  OEM_CHARSET
 };
 */
 
-    void message_alpha(double alpha);
-    void message_background(int back);
-    void message_button(int spr);
-    void message_button_font(string name, int size, int color, int style);
-    void message_caption(bool show, string str);
-    void message_input_color(int col);
-    void message_input_font(string name, int size, int color, int style);
-    void message_mouse_color(int col);
-    void message_position(int x, int y);
-    void message_size(int w, int h);
-    void message_text_font(string name, int size, int color, int style); 
-	void message_text_charset(int type, int charset); 
-	
-	int show_message_ext(string str, string but1, string but2, string but3);
-  
-	bool show_question(string str);
-	inline bool action_if_question(string str)
-	{
-		return show_question(str);
-	}
-	
-	// IMPLEMENTS from widgets_mandatory:
-	// void show_error(string errortext, const bool fatal);
+  void message_alpha(double alpha);
+  void message_background(int back);
+  void message_button(int spr);
+  void message_button_font(string name, int size, int color, int style);
+  void message_caption(bool show, string str);
+  void message_input_color(int col);
+  void message_input_font(string name, int size, int color, int style);
+  void message_mouse_color(int col);
+  void message_position(int x, int y);
+  void message_size(int w, int h);
+  void message_text_font(string name, int size, int color, int style);
+  void message_text_charset(int type, int charset);
 
-	int get_color(int defcol, bool advanced = false);
-	
-	string get_open_filename(string filter, string fname, string caption="");
-	string get_save_filename(string filter, string fname, string caption="");
-	string get_directory(string dname, string caption="Select Folder");
-	string get_directory_alt(string message, string root, bool modern=false, string caption="Browse for Folder");
+  int show_message_ext(string str, string but1, string but2, string but3);
 
-	string get_login(string username, string password, string cap="");
-	string get_string(string message, string def, string cap="");
-	double get_number(string message, string def, string cap="");
```

Comment:
Ok, so I went digging for the reason behind why `get_number` ever existed in ENIGMA when it didn't in GM. I tracked it down to before I got here, and it seems to be that `get_number` returned float and `get_integer` returned integers. You should ask Josh about this one, because that's still useful to have a dialog that accepts floats.

https://github.com/enigma-dev/enigma-dev/blob/04825003a9d2c420a960baa8978c8ee5f1e49ce7/ENIGMAsystem/SHELL/Widget_Systems/Win32/dialogs.cpp#L105
https://github.com/enigma-dev/enigma-dev/blob/04825003a9d2c420a960baa8978c8ee5f1e49ce7/ENIGMAsystem/SHELL/Widget_Systems/Win32/dialogs.cpp#L115

You can see here in GM5 and GM8 that `get_number` was not a function:
![GM5 get_number syntax error](https://user-images.githubusercontent.com/3212801/43256512-e6b79958-909a-11e8-8162-a72642dd532c.png)
![GM8 get_number syntax error](https://user-images.githubusercontent.com/3212801/43256475-cdaa3f6a-909a-11e8-8cc0-b237a0ed4430.png)

Specifically ask Josh because GMSv1.4 lets you input floats/doubles with `get_integer` so perhaps it should just return var. I don't know, ask him.
```gml
show_message(string(get_integer("hello", "55")));
show_message(string(get_integer("hello", "55f")));
```
### "0.457" becomes "0.46"
![GMSv1.4 get_integer accepting floats](https://user-images.githubusercontent.com/3212801/43256719-a124c414-909b-11e8-968c-fdad10a0eed3.png)
### "55f" becomes "55"
![GMSv1.4 get_integer not accepting letters](https://user-images.githubusercontent.com/3212801/43256694-88578520-909b-11e8-874f-00db450b26b2.png)

Reply:
In GM 8.1 it does not return an integer. Just like GMS 1.4 and 2, it returns a double. There's nothing for me to discuss with Josh. Everything is working how it is supposed to. The bottom line - get_number is the same thing as what enigma's original get_integer should've been. So I say get rid of get_number because we don't need a function that return an int, get_integer never did that in GM.

Reply:
Yes, that's the issue, that GM8.1 didn't have a function to return a correctly typed integer. ENIGMA is actual C++, which lets you work with types (this is a GM augmentation, not a clone!). Hence, that would explain why Josh added `get_integer` to begin with (I think he wrote this original widget system, not sure).

Reply:
I am happy to maintain the `get_number`/`get_integer` distinction in the name of sanity. A find/replace on the GMK file could patch over the difference (and I'm planning a more complicated compatibility layer to handle Dailly's other crimes).